src/ni/vsbuild/Pipeline.groovy

```diff
@@ -242,6 +249,31 @@ class Pipeline implements Serializable {
       }
    }
 
+   private void sendNotification() {
+      def pipelineResult = PipelineStatus.getResult(script)
+      if (pipelineResult == PipelineResult.SUCCESS || pipelineResult == PipelineResult.ABORTED) {
+         // Don't spam notifications if the build is consistently successful
+         // or if the build was manually aborted
+         return
+      }
+
+      if (!(script.env.BRANCH_NAME.startsWith("master") || script.env.BRANCH_NAME.startsWith("release"))) {
```

Comment:
PR description (and comment below) says "will only send notifications for builds on the `mainline` and `release` branches", but here we specify `startsWith("master")` or `startsWith("release")`.

1. It should be an equality check for the master branch, or we will include branches like `master-2`.
1. We should be consistent about which branches are included.
1. This can be out-of-scope for this PR, but we might want to allow defining a regex for the branch names in the build.toml file. This would make transitioning off the `master` branch as the default branch easier.

Reply:
Created a default regex for branches that matches "master" exactly and "release" followed by some non-word character. Enabled specification of this regex in `build.toml` and tested a build that passed and a build that failed to make sure I got the correct notifications. Also verified if the regex is not specified, I get no notification if it's not one of the default branches.